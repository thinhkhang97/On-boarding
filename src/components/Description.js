/// Trouble with font family.

import * as React from 'react';
import { Text } from 'react-native';

type Props = {
  style?: object
};

class Description extends React.Component<Props> {
  static defaultProps = {
    style: { fontSize: 16, color: '#7f8da2' }
  };
  render() {
    return <Text style={this.props.style}>{this.props.children}</Text>;
  }
}

export default Description;
