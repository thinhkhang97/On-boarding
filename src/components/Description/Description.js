// @flow
/// Trouble with font family.

import * as React from 'react';
import { View, Text } from 'react-native';
import type { ____ViewStyleProp_Internal as Style } from 'react-native/Libraries/StyleSheet/StyleSheetTypes';

type Props = {
  style?: Style,
  children?: React.Node,
};

class Description extends React.Component<Props> {
  static defaultProps = {
    style: {},
  };
  render() {
    const { style, children } = this.props;
    return <Text style={[{ fontSize: 16, color: '#7f8da2' }, style]}>{children}</Text>;
  }
}

export default Description;
