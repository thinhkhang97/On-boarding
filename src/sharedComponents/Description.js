/// Trouble with font family.

import * as React from 'react';
import { Text } from 'react-native';

type Props = {
  style?: object
};

class Description extends React.Component<Props> {
  static defaultProps = {
    style: {}
  };
  render() {
    return (
      <Text style={[{ fontSize: 16, color: '#7f8da2' }, this.props.style]}>
        {this.props.children}
      </Text>
    );
  }
}

export default Description;
