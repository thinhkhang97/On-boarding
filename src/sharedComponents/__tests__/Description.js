import * as React from 'react';
import renderer from 'react-test-renderer';
import Description from '../Description';

test('Render button correctly', ()=>{
    const testDescription = renderer.create(<Description/>).toJSON();
    expect(testDescription).toMatchSnapshot();
})