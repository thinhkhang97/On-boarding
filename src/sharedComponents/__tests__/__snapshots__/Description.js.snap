// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render button correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#7f8da2",
        "fontSize": 16,
      },
      Object {},
    ]
  }
/>
`;
